doc/HOWTO/howto_readState.md

```diff
@@ -0,0 +1,165 @@
+# HOWTO: Read State from the RPC
+
+## Goals 
+In [ADR1](../ADR/ADR_001.md) we described how we wanted to use an intermediate representation of State, to work with within Solarkraft. 
+
+This HOWTO breaks down the various steps required to programmatically extract State in Javascript/Typescript, by using Soroban RPC calls.
+
+## Process
+
+We will be using [this contract][contract] in our example. Its internal state is composed of a single `COUNTER` of type `u32`.
+
+To start with, we will need the deployed contract's ID. To quote from [this guide][guide]:
+
+> When you deploy a contract, first the code is "installed" (i.e., it is uploaded onto the blockchain). This creates a LedgerEntry containing the Wasm byte-code, which is uniquely identified by its hash (that is, the hash of the uploaded code itself). 
+
+Next, we need to identify the value, or set of values, we wish to read. In our case, this is a single field, named `CONTRACT`. 
+
+For each of these values, we first need to determine under which key they are stored in the ledger. This involves computing an XDR representation.
+
+The following snippet shows a function, which can be used to compute this for any contract ID and any field name (using the `stellar-sdk`):
+```js
+const getLedgerKeySymbol = (contractId, symbolText) => {
+  const ledgerKey = xdr.LedgerKey.contractData(
+    new xdr.LedgerKeyContractData({
+      contract: new Address(contractId).toScAddress(),
+      key: xdr.ScVal.scvSymbol(symbolText),
+      durability: xdr.ContractDataDurability.persistent(),
+    }),
+  );
+  return ledgerKey.toXDR("base64");
+};
+```
+
+We need to call `getLedgerKeySymbol(contractID, "COUNTER")`. If we had a contract with multiple fields, we would call this method for each of them:
+```
+k = getLedgerKeySymbol(contractID, "COUNTER")
+k_foo = getLedgerKeySymbol(contractID, "FOO")
+k_bar = getLedgerKeySymbol(contractID, "BAR")
+```
+
+With the key, or set of keys, in hand, we now invoke the RPC, to querry the current values of the fields these keys represent.
+
+In Javascript, querying the RPC is done as follows:
+```js
+let res = await fetch(NET, {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json',
+  },
+  body: JSON.stringify(requestBody),
+})
+let json = await res.json()
+```
+where `NET` is the Soroban network being queried, for example 
+`'https://soroban-testnet.stellar.org'`, and `requestBody` is the specific request we're making of the network.
+
+Since we aim to read a ledger entry, our request will use the `getLedgerEntries` method, and `requestBody` will look like this:
```

Comment:
I still don't think we should limit ourselves to use `getLedgerEntries`, given that it only returns the current state of a contract.

afaiu, the solution for requesting a specific ledger may look very different.